salt/modules/cp.py

```diff
@@ -267,12 +268,19 @@ def get_file(
     .. note::
         It may be necessary to quote the URL when using the querystring method,
         depending on the shell being used to run the command.
+
+    If ``saltenv`` is not explicitely defined, the one from minion configuration
+    is used.
     """
     (path, dest) = _render_filenames(path, dest, saltenv, template, **kwargs)
 
     path, senv = salt.utils.url.split_env(path)
     if senv:
         saltenv = senv
+    if not saltenv:
+        saltenv = salt.utils.state.get_sls_opts(__opts__, **kwargs).get(
+            "saltenv", "base"
+        )
```

Comment:
**suggestion**: this could be rewritten like:

```suggestion
    saltenv = senv if senv else salt.utils.state.get_sls_opts(__opts__, **kwargs).get("saltenv", "base")
```